python/src/robyn/data/validation/mmmdata_validation.py

```diff
@@ -15,46 +17,176 @@ def check_missing_and_infinite(self) -> ValidationResult:
         Check for columns with no variance in the input dataframe.
         
         :return: A list of column names with no variance.
         """
-        raise NotImplementedError("Not yet implemented")
+        dt_input = self.mmm_data.data
+        
+        no_variance_cols = dt_input.columns[dt_input.nunique() == 1].tolist()
+        
+        error_details = {}
+        error_message = ""
+        
+        if no_variance_cols:
+            error_details['no_variance'] = no_variance_cols
+            error_message = f"There are {len(no_variance_cols)} column(s) with no-variance: {', '.join(no_variance_cols)}. Please remove the variable(s) to proceed."
+        
+        return ValidationResult(
+            status=not bool(error_details),
+            error_details=error_details,
+            error_message=error_message
+        )
+    
 
     def check_variable_names(self) -> ValidationResult:
         """
         Check variable names for duplicates and invalid characters.
         
         :return: A dictionary with keys 'duplicates' and 'invalid', each containing a list of problematic column names.
         """
-        raise NotImplementedError("Not yet implemented")
+
+        mmmdata_spec = self.mmm_data.mmmdata_spec
+
+        # Collect all variable names to check
+        vars_to_check = [
+            mmmdata_spec.dep_var,
+            mmmdata_spec.date_var,
+            mmmdata_spec.context_vars,
+            mmmdata_spec.paid_media_spends,
+            mmmdata_spec.organic_vars
+        ]
+        vars_to_check = [var for var in vars_to_check if var is not None]
+        vars_to_check = [item for sublist in vars_to_check for item in (sublist if isinstance(sublist, list) else [sublist])]
+
+        # Check for duplicates
+        duplicates = [var for var in set(vars_to_check) if vars_to_check.count(var) > 1]
+
+        # Check for invalid characters (spaces)
+        invalid = [var for var in vars_to_check if ' ' in var]
+
+        #prepare error messages
+        error_details = {}
+        error_message = ""
+        if duplicates:
+            error_details['duplicates'] = duplicates
+            error_message += "Duplicate variable names present. "
+        if invalid:
+            error_details['invalid'] = invalid
+            error_message += "Invalid variable names present. "
+
+
+        return ValidationResult(
+            # status is true if error_details is empty
+            status=not error_details,
+            error_details=error_details,
+            error_message=error_message
+        )
+    
     
     def check_date_variable(self) -> ValidationResult:
         """
         Checks if the date variable is correct.
         
         :return: True if the date variable is valid, False otherwise.
         """
-        raise NotImplementedError("Not yet implemented")
+        mmmdata_spec = self.mmm_data.mmmdata_spec
+        dt_input = self.mmm_data.data
+        date_var = mmmdata_spec.date_var
```

Comment:
default value for date_var is "auto". Either define what "auto" means and update code or just remove "auto" as default value so that it becomes mandatory for user to specify the date variable name.

Reply:
Good call. In R, the code automatically checks for a column that matches is.Date. In python Ill have throw an  error for now.